### DISCUSSION TITLE
PhoneBlock-Diskussionsforum

### DISCUSSION THREAD

**Discussion by haumacher** [haumacher/phoneblock#21] [Announcements]
## Willkommen beim PhoneBlock-Diskussionsforum

Diskussionen ist ein Ort, um mit anderen Mitgliedern der PhoneBlock-Community in Kontakt zu treten. Hier kannst Du:

* Fragen stellen
* Ideen mit anderen Community-Mitgliedern austauschen
* Erfahrungen mit PhoneBlock berichten
* Andere willkommen heißen

Um loszulegen, stell Dich in einem Kommentar unten vor und erzähl uns von Deinen Erfahrungen mit PhoneBlock.

**Comment by mrtwiste** [comment]
Hallo, bei meiner 7430, Firmware 07.29 wird der Sync der Liste ohne Fehlermeldung durchgeführt. 
Dennoch bleibt die Liste leer und im Ereignislog erscheint die folgende Meldung: " Maximale Anzahl Rufnummern im Telefonbuch überschritten"
Ist diese Box zu alt? Bzw. hat die ein Speicherproblem? 
Gruß

**Reply by haumacher** [reply]
Kann sein, dass Deine Box keine so großen Telefonbücher verträgt. Probier doch mal Dich auf [der PhoneBlock-Webseite einzuloggen](https://phoneblock.haumacher.de/phoneblock/login.jsp) und die "Maximale Blocklist-Größe" auf 900 zu stellen. Vielleicht kannst Du PhoneBlock dann trotzdem mit einer kleineren Blocklist nutzen.

**Comment by artfarmer** [comment]
Phoneblock via Fritzbox: Nur eine Telefonbuch-Sperre möglich.

Es lässt sich in der Fritzbox (bei mir 7490 mit FritzOS 7.59) offenbar **nur ein Telefonbuch komplett sperren**, anderenfalls kommt eine Fehlermeldung. Erst nachdem ich mein Telefonbuch "Gesperrte Nummern" von der Sperrliste entfernt habe, konnte ich das Telefonbuch "Blocklist" sperren – danach habe ich versucht "Gesperrte Nummern" wieder zu sperren – nope.

Dies ist nur als Hinweis gedacht. Hab schon an AVM geschrieben.
Sollte ich falsch liegen, bitte hier schreiben – dann müsste ich meine Konfiguration nochmal checken. ;)

**Comment by sristow646** [comment]
Hallo Zusammen, ich habe mir über KI einige Sätze für den AB erzeugt und auch sprechen lassen. In der Anleitung steht "mit dem Tool ConversionInitializer initialisiert werden" wie mache ich das?. Mit Java habe ich bisher nichts zu tun gehabt, also verzeiht mir meine Unwissenheit. Ich habe die Dateien mit Audacity in einzelne Dateien Umwandeln lassen, aber wenn ich den AB teste kommt nur gequicke und rauschen. Eventuell habe ich ja auch nur ne falsche sample rate genommen.  Wenn es dann mal funktioniert kann ich die Sprachdateien auch gerne der Allgemeinheit zur Verfügung stellen

**Reply by geimist** [reply]
> ich habe mir über KI einige Sätze für den AB erzeugt und auch sprechen lassen

Was hast du da verwendet?

**Reply by sristow646** [reply]
Ich habe in Chatgpt für die jeweilige Situation Sätze erstellen lassen mit folgenden Prompt "Welche 20 Begrüßung am Telefon ohne meinen Namen zu nennen, es darf kein JA drin vorkommen"  Es kommt dann Sätze wie :
Herzlich willkommen, wie kann ich Ihnen helfen?
Grüß Gott, wie kann ich Ihnen weiterhelfen?
Einen schönen Tag wünsche ich, wie kann ich Ihnen zur Seite stehen?
Hallo, wie kann ich Ihre Anfrage beantworten?
Guten Morgen, was kann ich für Sie tun?
Hallo und willkommen, wie darf ich Ihnen behilflich sein?

Mit https://elevenlabs.io/app/speech-synthesis/text-to-speech habe ich dann die Sätze eingefügt:
Hallo, wie kann ich Ihre Anfrage beantworten?"<break time="2s"/>"
Guten Morgen, was kann ich für Sie tun?"<break time="2s"/>"
Hallo und willkommen, wie darf ich Ihnen behilflich sein?"<break time="2s"/>"

Das break Time fügt dann jeweils 2 sec pause zwischen den Sätzen ein.

Ich werd das mal mit Deinem Script testen.
Danke Schon mal

**Reply by geimist** [reply]
Die text-to-speech Seite ist super. Die fehlte mir noch. Danke 🤩

**Comment by geimist** [comment]
@sristow646 
Ich habe mir ein kleines Bashskript für [ffmpeg](https://www.ffmpeg.org/download.html) erstellt. 
Einzeln geht es auch mit dieser Zeile: `ffmpeg -i "Quelldatei" -c:a pcm_alaw -ar 8000 -ac 1 "Zieldatei"`

```bash
#!/bin/bash

# Prüfen, ob ein Ordner als Parameter übergeben wurde
if [ -z "$1" ]; then
    echo "Bitte einen Ordner als Parameter angeben."
    exit 1
fi

INPUT_DIR="$1"

# Prüfen, ob der angegebene Ordner existiert
if [ ! -d "$INPUT_DIR" ]; then
    echo "Der Ordner '$INPUT_DIR' existiert nicht."
    exit 1
fi

# Funktion, um eindeutigen Dateinamen zu erstellen
generate_unique_filename() {
    local dir="$1"
    local base_name="$2"
    local extension="$3"
    local counter=1
    local new_name="${base_name}.${extension}"

    while [ -e "$dir/$new_name" ]; do
        new_name="${base_name}_$counter.${extension}"
        ((counter++))
    done

    echo "$new_name"
}

# Schleife durch alle Audiodateien im Ordner
for file in "$INPUT_DIR"/*.{mp3,flac,wav,m4a,ogg}; do
    # Prüfen, ob Datei existiert
    [ -e "$file" ] || continue

    # Extrahiere Dateinamen ohne Erweiterung
    base_name=$(basename "$file" | sed 's/\.[^.]*$//')
    output_file=$(generate_unique_filename "$INPUT_DIR" "$base_name" "wav")

    echo "Konvertiere: $file -> $INPUT_DIR/$output_file"

    # Konvertierung mit ffmpeg
    ffmpeg -i "$file" -c:a pcm_alaw -ar 8000 -ac 1 "$INPUT_DIR/$output_file"
done

echo "Konvertierung abgeschlossen."
```

**Reply by sristow646** [reply]
Danke, das mit dem Script funktioniert perfekt, als i tüpfelchen wäre es noch schöner wenn die Ausgabe in einem anderen Ordner gespeichert werden würde. Sollte aber leicht zu ändern sein.  Auch würde ich gerne eine etwas ausführlichere Anleitung schreiben, darf ich da Dein Script (unter Nennung Deines Namens) mit einfügen?

**Reply by geimist** [reply]
Freut mich, dass es dir hilft. Selbstverständlich kannst du damit machen, was du möchtest. Und wenn du es für die Allgemeinheit optimierst, um so besser. 🥳

**Comment by haumacher** [comment]
Das mit der Text-To-Speech-KI hat mir keine Ruhe gelassen - ich habe mal eine Demo-Konversation damit erzeugt: https://github.com/haumacher/phoneblock/tree/master/phoneblock-ab/conversation

Klingt vielleicht nicht 100% überzeugend aber für die SPAMer erst mal gut genug :-)

@geimist - ich habe Dein Script als Basis für die Konvertierung genommen, Danke! Meine Version liegt jetzt hier: https://github.com/haumacher/phoneblock/blob/master/phoneblock-ab/bin/convert-audio.sh

Ich habe noch hinzugefügt, dann man einen ganzen Satz von Ordnern auf einmal konvertieren kann und dass die für PhoneBlock notwendige Ordnerstruktur automatisch generiert wird.

**Reply by sristow646** [reply]
Ich habe festgestellt, das man bei einigen Stimmen das raushört ob es eine KI ist oder nicht. Je nachdem ob Du für das Text to Speech die Bezahlvariante genutzt hast sind die Ergebnisse besser oder nicht.
Aber wie Du ja schreibst die Files reichen für die SPAMer ;-)
Ich hatte mir das günstigste Abo für einen Monat dort geleistet. Die Files hab ich mal in mein Github getan: https://github.com/sristow646/conversation-sample

Da ich den Phoneblocker-AB etwas anders nutze passen meine Ansagetexte nicht bei Dir, aber wenn Du die trotzdem haben möchtest kannst Du die gerne nutzen.

Es ist immer der gleiche Text nur unterschiedliche Stimmen, es soll für die ja nicht Langweilig werden :-)

**Reply by haumacher** [reply]
Ja cool, es gibt ja auch den Wunsch (https://github.com/haumacher/phoneblock/issues/122) dass man den Ansagetext wählen kann - also eine klare Anrufabsage machen kann, statt den Anrufer "auf den Arm" zu nehmen. Mal sehen, wann ich dazu komme, diesen Wunsch umzusetzen....

**Comment by Kaschtelator** [comment]
Jetzt hab ich auch mal nen Docker Container aufgesetzt aber eine Sache fehlt mir 
so ne kleine Seite wo man sehn kann wie viele Spamer es versucht haben 
und für den Container eine einfachere art einzustellen das man aufnahmen will 
das wäre der Hit aber sonst läuft und macht weiterhin spaß ^^

**Reply by geimist** [reply]
> so ne kleine Seite wo man sehn kann wie viele Spamer es versucht haben

Mir reicht die Info in der FritzBox (man sieht ja, wann der AB rangegangen ist). 
Oder das gemappte Verzeichnis mit den Aufnahmen.


> für den Container eine einfachere art einzustellen das man aufnahmen will

Dafür gibt es eine Umgebungsvariable:
```YAML
…
    environment:
      # Das Verzeichnis, in dem die Aufnahmen gespeichert werden sollen. Wenn kein Verzeichnis angegeben wird, werden keine Audios aufgezeichnet.
#     - RECORDINGS=none
      - RECORDINGS=/opt/phoneblock/recordings
```

**Reply by Kaschtelator** [reply]
Ja gut recordings jetzt in einen extra Ordner zu machen war schon hilfreich.
Danke 
Aber für mich wäre es halt ultra praktisch wenn im docker Container noch ein kleiner Webserver läuft der wie die "Phoneblock ab" Seite aufgebaut wäre.
Und vielleicht sofern man es einstellt auch schauen kann was für recordings vorhanden sind.
Wenn man ja schon selbst hostet.

So mein Gedanke zur Faulheit 😁 bzw Bequemlichkeit.

**Reply by haumacher** [reply]
Die Idee ist gut, allerdings steht sie nicht ganz oben auf meiner Prioritätenliste. Eigentlich ist die Mission von PhoneBlock eine Anti-Spam-Lösung bereitzustellen, die massentauglich ist. Und das ist das Hosten von Docker-Containern eher nicht. Aber wenn sich jemand findet, der den Standalone-AB um eine Spring-Boot-App mit kleiner Oberfläche erweitern möchte, dann sind PRs natürlich willkommen.
